app/src/sharedTest/java/org/oppia/app/profile/AdminAuthActivityTest.kt

```diff
@@ -132,6 +135,115 @@ class AdminAuthActivityTest {
     }
   }
 
+  @Test
+  fun testAdminAuthActivity_inputTextPreservedOnRotation() {
+    launch<AdminAuthActivity>(
+      AdminAuthActivity.createAdminAuthActivityIntent(
+        context,
+        "12345",
+        -10710042
+      )
+    ).use {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(
+        typeText("12345"),
+        closeSoftKeyboard()
+      )
+      it.onActivity {
+        it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        assert(it.requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+      }
+      onView(withId(R.id.submit_button)).check { view, noViewFoundException ->
+        view.background = context.getDrawable(R.drawable.state_button_primary_background)
+      }
+      it.onActivity {
+        it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        assert(it.requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_PORTRAIT)
+      }
+      onView(withId(R.id.submit_button)).check { view, noViewFoundException ->
+        view.background = context.getDrawable(R.drawable.state_button_primary_background)
+      }
+    }
+  }
+
+  @Test
+  fun testAdminAuthActivity_buttonStateIsPreservedOnRotation() {
+    launch<AdminAuthActivity>(
+      AdminAuthActivity.createAdminAuthActivityIntent(
+        context,
+        "12345",
+        -10710042
+      )
+    ).use {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(
+        typeText("12345"),
+        closeSoftKeyboard()
+      )
+      it.onActivity {
+        it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        assert(it.requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+      }
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).check(
+        matches(
+          withText("12345")
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testAdminAuthActivity_inputCorrectPasswordLandscape_opensAddProfileActivity() {
+    launch<AdminAuthActivity>(
+      AdminAuthActivity.createAdminAuthActivityIntent(
+        context,
+        "12345",
+        -10710042
+      )
+    ).use {
+      it.onActivity {
+        it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        assert(it.requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+      }
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(
+        typeText("12345"),
+        closeSoftKeyboard()
+      )
+      onView(withId(R.id.submit_button)).perform(scrollTo(), click())
+      intended(hasComponent(AddProfileActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testAdminAuthActivity_inputIncorrectPasswordLandscape_checkError() {
```

Comment:
Also update this text case to check error preservation, meaning:

1. Open aactiivty
2. Enter some incorrect pin
3. Click submit 
This will result in some error message
4. Rotate to landscape
5. Check if the error is still visible or not. (If visible then pass else fail)


Note: currently such test case will fail as the error text disappears after rotation.

You will need to use `onSaveInstanceState` in the activity/fragment/view to preserve this error. Feel free to ask for help if you get stuck.

Reply:
@rt4914  Should I update this test only, or make one more test for this?

Reply:
Modify this test case only.

Reply:
Done